src/function/scalar/string/like.cpp

```diff
@@ -6,6 +6,16 @@ using namespace std;
 
 namespace duckdb {
 
+namespace {
+
+std::string tolower(string_t& str) {
+	std::string s(str.GetData(), str.GetSize());
+	std::transform(s.begin(), s.end(), s.begin(), ::tolower);
```

Comment:
I believe the system function ::tolower only handles ASCII characters. `ILIKE` should also handle non-ascii characters correctly. For that reason, could you switch this to using system `tolower` function (found in [caseconvert.cpp](https://github.com/cwida/duckdb/blob/11e51619d8402c649cdb45011944a1ca504060f9/src/function/scalar/string/caseconvert.cpp))?

Reply:
The std::tolower function can also take the locale of the db (not sure if we have that somewhere) in to account while converting the case maybe that's something that we can use

Reply:
We want to avoid using the standard library locales, there are many potential issues with locales needing to be installed on the host system and queries having different results depending on system locales. std::tolower is also problematic because of std::string being used, which introduces large performance penalties. Better to just use our own lowercase function here.

Reply:
Okay that makes a lot of sense, thanks for clearing it up for me.